### PR TITLE
chore: extract move gifs module

### PR DESCRIPTION
**Pull request by igor-sirotin**
## Description

Extracted a GIFs module.
It's now created a single time and reused for all chats.

Fixes https://github.com/status-im/status-desktop/issues/9952

### CODE REVIEW

File: src/app/modules/main/gifs/io_interface.nim [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  raise newException(ValueError, "No implementation available")
+
+method isLoaded*(self: AccessInterface): bool {.base.} =
+  raise newException(ValueError, "No implementation available")
+
+method viewDidLoad*(self: AccessInterface) {.base.} =
+  raise newException(ValueError, "No implementation available")
+
+# method getModuleAsVariant*(self: AccessInterface): QVariant {.base.} =
+#   raise newException(ValueError, "No implementation available")
```
Comment (jrainville) [inline src/app/modules/main/gifs/io_interface.nim:19]: We can remove this

**Review by jrainville — APPROVED** [review]
Looks good!

File: src/app/boot/app_controller.nim [hunk 2]
```diff
@@ -467,7 +466,6 @@ proc load(self: AppController) =
     self.chatService,
     self.communityService,
     self.messageService,
-    self.gifService,
```
Comment (kounkou) [inline src/app/boot/app_controller.nim:470]: Hi there, thanks for the changes, do you mind explaining what is the new architecture design? What are some triggers and consequences of this change please? Does that mean for instance that the main module will be slower to start?

**Comment by kounkou** [comment]
Hey, just curious, could you please list the different use cases you went through (in your mind) of ensuring your change is complete?

**Review by kounkou — APPROVED** [review]
Thanks for the refactor, no blocking comments, just curiosity question. I wonder if for instance in the absence of Jo, this could have been detected based on what the user sees during the app execution...

**Comment by igor-sirotin** [comment]
@kounkou  Well it's not that of a major change 😄 
This module is only used in for the GIF popup. So we don't need to create it for each chat, just once is enough. 

> Hey, just curious, could you please list the different use cases you went through (in your mind) of ensuring your change is complete?

Just opened the GIF popup and tested that it's working as should 🙂 

> What are some triggers and consequences of this change please?

It really just a refactoring, that improves the code logic and should improve the performance a bit.

> Does that mean for instance that the main module will be slower to start?

In theory it should only be faster, as we're creating it less times.

**Pull request merged**